### ISSUE TITLE
Using --transparent-background "0.0,0.0,0.0" crashes while pruning

### ISSUE THREAD

**Issue by nitthilan** [facebookresearch/NSVF#24]
**Describe the bug**
I am trying to run this code for a custom dataset. The dataset is as follows. 
https://drive.google.com/drive/folders/1p6L5FVMrGzz3wdOBZwlo7ZCjS-AfsDfY?usp=sharing

And I use the following command to run 
export DATASET="../neural_sparse_voxel_field/data/pixologic/brian_pape_1/nsvf/"
export SAVE="./brian_pape_1_ckpt/"
mkdir brian_pape_1_ckpt
rm -rf brian_pape_1_ckpt/*

CUDA_VISIBLE_DEVICES=1,2,4,5,6,7,8 python -u train.py ${DATASET} \
    --user-dir fairnr \
    --task single_object_rendering \
    --train-views "0..15" --view-resolution "562x750" \
    --max-sentences 1 --view-per-batch 1 --pixel-per-view 2048 \
    --no-preload \
    --sampling-on-mask 1.0 --no-sampling-at-reader \
    --valid-views "0..8" --valid-view-resolution "281x375" \
    --valid-view-per-batch 1 \
    --transparent-background "0.0,0.0,0.0" --background-stop-gradient \
    --arch nsvf_base \
    --initial-boundingbox ${DATASET}/bbox.txt \
    --use-octree \
    --raymarching-stepsize-ratio 0.125 \
    --discrete-regularization \
    --color-weight 128.0 --alpha-weight 1.0 \
    --optimizer "adam" --adam-betas "(0.9, 0.999)" \
    --lr 0.001 --lr-scheduler "polynomial_decay" --total-num-update 150000 \
    --criterion "srn_loss" --clip-norm 0.0 \
    --num-workers 0 \
    --seed 2 \
    --save-interval-updates 200 --max-update 150000 \
    --virtual-epoch-steps 500 --save-interval 1 \
    --half-voxel-size-at  "500,2500,7500" \
    --reduce-step-size-at "500,2500,7500" \
    --pruning-every-steps 250 \
    --keep-interval-updates 5 --keep-last-epochs 5 \
    --log-format simple --log-interval 1 \
    --save-dir ${SAVE} \
    --tensorboard-logdir ${SAVE}/tensorboard \
    | tee -a $SAVE/train.log

<img width="1064" alt="Screenshot 2020-10-27 at 9 59 25 PM" src="https://user-images.githubusercontent.com/4926564/97395018-3e5cc300-18a1-11eb-9cee-c50c80e65268.png">

The above error when the first stage of pruning happens. Looks like the number of nodes after pruning seems to be zero. I am not sure how this can happen. Any ideas where I have to look for debugging?

The loss gradually decreases to around 10-30 and the validation PSNR was around 19 and ssim 0.73. So I assumed some voxels should have good values. This was not the case with 
--transparent-background "1.0,1.0,1.0". In this case, though the ssim is stuck at 0.73 and slowly rises but the training loss is around 40-50. 

Any ideas and directions to debug this??

**Comment by MultiPath** [comment]
Hi, it seems that your data is transparent? 
So the ``--transparent-background`` is simply given a background color.
I have to try on my side with your data to see how it looks.

**Comment by MultiPath** [comment]
It would be helpful if you have logs or tensorboard logs?

**Comment by nitthilan** [comment]
Uploaded the logs and tensorboard logs in the shared folder under brian_pape_1_ckpt folder.
Check it out

**Comment by nitthilan** [comment]
Hi,
I think I found the issue. It was with respect to the input parameters. I had kept the pruning-every_step value to 250 which is small. Looks like during the initial training phase, only transparent areas are sampled and so the neural network trains to a all zero output case which provides a good loss value.  Due to this the pruning returns that no voxels are valid. Increasing the pruning cycle removed this error. Closing this issue for now. Thanks for all the help
